monitoring/monitoring_internal_test.go

```diff
@@ -181,528 +192,543 @@ func TestPeriodicReport(t *testing.T) {
-	if err != nil {
-		t.Fatalf("Can't stop monitoring service: %s", err)
+	case <-time.After(duration * 2):
+		t.Fatal("Monitoring data timeout")
 	}
 
-	_ = runContainerWait("service1", cmd1)
+	for i, item := range testData {
+		if item.alert != nil {
+			for _, expectedAlert := range item.alert {
+				found := false
 
-	_ = runContainerWait("service2", cmd2)
-}
-
-func TestTrafficLimit(t *testing.T) {
-	sendDuration := 2 * time.Second
-
-	sender := &testSender{}
-
-	monitor, err := New(
-		&config.Config{
-			WorkingDir: ".",
-			Monitoring: config.Monitoring{
-				SendPeriod: config.Duration{Duration: sendDuration},
-				PollPeriod: config.Duration{Duration: 1 * time.Second},
-			},
-		},
-		sender, networkManager)
-	if err != nil {
-		t.Fatalf("Can't create monitoring instance: %s", err)
-	}
-	defer monitor.Close()
-
-	if err := networkManager.SetTrafficPeriod(MinutePeriod); err != nil {
-		t.Errorf("Can't set traffic period: %s", err)
-	}
-
-	// wait for beginning of next minute
-	time.Sleep(time.Duration((60 - time.Now().Second())) * time.Second)
-
-	cmd1, err := runContainerCmd(path.Join(tmpDir, "service1"), "service1", 300, 300)
-	if err != nil {
-		t.Fatalf("Can't start service: %s", err)
-	}
-
-	// Wait while .ip amd .pid files are created
-	time.Sleep(1 * time.Second)
-
-	ipAddress, err := networkManager.GetServiceIP("service1", "default")
-	if err != nil {
-		t.Fatalf("Can't get service IP: %s", err)
-	}
-
-	err = monitor.StartMonitorService("service1",
-		ServiceMonitoringConfig{
-			ServiceDir: "tmp/service1",
-			IPAddress:  ipAddress,
-			UID:        5001,
-			GID:        5001,
-		})
-	if err != nil {
-		t.Fatalf("Can't start monitoring service: %s", err)
-	}
-
-	if err := runContainerWait("service1", cmd1); err == nil {
-		t.Error("Ping should fail")
-	}
-
-	err = monitor.StopMonitorService("service1")
-	if err != nil {
-		t.Fatalf("Can't stop monitoring service: %s", err)
-	}
-
-	// wait for beginning of next minute
-	time.Sleep(time.Duration((60 - time.Now().Second())) * time.Second)
-
-	// Start again
-
-	if cmd1, err = runContainerCmd(path.Join(tmpDir, "service1"), "service1", 2000, 2000); err != nil {
-		t.Fatalf("Can't start service: %s", err)
-	}
-
-	if ipAddress, err = networkManager.GetServiceIP("service1", "default"); err != nil {
-		t.Fatalf("Can't get service IP: %s", err)
-	}
+				for _, receivedAlert := range alertMap {
+					if expectedAlert.Payload == receivedAlert.Payload {
+						found = true
+					}
+				}
 
-	// Wait while .ip amd .pid files are created
-	time.Sleep(1 * time.Second)
+				if !found {
+					t.Error("Incorrect system alert payload")
+				}
+			}
+		}
 
-	err = monitor.StartMonitorService("service1",
-		ServiceMonitoringConfig{
-			ServiceDir: "tmp/service1",
-			IPAddress:  ipAddress,
-			UID:        5001,
-			GID:        5001,
-		})
-	if err != nil {
-		t.Fatalf("Can't start monitoring service: %s", err)
+		if err := monitor.StopInstanceMonitor(fmt.Sprintf("instance%d", i)); err != nil {
```

Comment:
Why do you need to stop instance monitoring inside this loop?

Reply:
because above in the loop we call start monitoring, and below we check that there is no instance monitoring data